sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -193,13 +193,32 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
    * This is only used for Common Table Expressions.
    */
   override def visitNamedQuery(ctx: NamedQueryContext): SubqueryAlias = withOrigin(ctx) {
-    val subQuery: LogicalPlan = plan(ctx.query).optionalMap(ctx.columnAliases)(
+    val logicalPlan = Option(ctx.query).map(plan).orElse(
+      Option(ctx.ddlStatementForQuery).map(visitDdlStatementForQuery)).get
```

Comment:
nit: we can call `visitDdlQuery` and don't need to create `visitDdlStatementForQuery`